func TestZipReader(t *testing.T) {}

Out of the box TestZipReader doens't seem to be working

```% go test
--- FAIL: TestZipReader (0.01s)
        zipreader_test.go:75: Testing zipped reading for test_files/polyline
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/polyline read was wrong. Wanted 2, got 1.
        zipreader_test.go:75: Testing zipped reading for test_files/multipoint
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/multipoint read was wrong. Wanted 1, got 0.
        zipreader_test.go:75: Testing zipped reading for test_files/polylinez
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/polylinez read was wrong. Wanted 2, got 1.
        zipreader_test.go:75: Testing zipped reading for test_files/multipatch
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/multipatch read was wrong. Wanted 1, got 0.
        zipreader_test.go:75: Testing zipped reading for test_files/polygonz
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/polygonz read was wrong. Wanted 1, got 0.
        zipreader_test.go:75: Testing zipped reading for test_files/polylinem
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/polylinem read was wrong. Wanted 2, got 1.
        zipreader_test.go:75: Testing zipped reading for test_files/polygonm
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/polygonm read was wrong. Wanted 1, got 0.
        zipreader_test.go:75: Testing zipped reading for test_files/point
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/point read was wrong. Wanted 3, got 2.
        zipreader_test.go:75: Testing zipped reading for test_files/pointz
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/pointz read was wrong. Wanted 3, got 2.
        zipreader_test.go:75: Testing zipped reading for test_files/multipointz
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/multipointz read was wrong. Wanted 1, got 0.
        zipreader_test.go:75: Testing zipped reading for test_files/multipointm
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/multipointm read was wrong. Wanted 1, got 0.
        zipreader_test.go:75: Testing zipped reading for test_files/polygon
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/polygon read was wrong. Wanted 1, got 0.
        zipreader_test.go:75: Testing zipped reading for test_files/pointm
        zipreader_test.go:64: Error when iterating over the shapes: Error while reading next shape: EOF
        reader_test.go:220: Number of shapes for test_files/pointm read was wrong. Wanted 3, got 2.
FAIL
exit status 1
FAIL    github.com/gertcuykens/go-shp   0.029s
%```

--- a/errreader.go
+++ b/errreader.go
@@ -1,6 +1,9 @@
 package shp
 
-import "io"
+import (
+	"fmt"
+	"io"
+)
 
 // errReader is a helper to perform multiple successive read from another reader
 // and do the error checking only once afterwards. It will not perform any new
@@ -13,7 +16,7 @@ type errReader struct {
 
 func (er *errReader) Read(p []byte) (n int, err error) {
 	if er.e != nil {
-		return 0, er.e
+		return 0, fmt.Errorf("unable to read after previous error: %v", er.e)
 	}
 	n, er.e = er.Reader.Read(p)
 	er.n += int64(n)

--- a/sequentialreader.go
+++ b/sequentialreader.go
@@ -148,7 +148,13 @@ func (sr *seqReader) Next() bool {
 	sr.num = num
 	sr.shape = newShape(shapetype)
 	sr.shape.read(er)
-	if er.e != nil {
+	switch {
+	case er.e == io.EOF:
+		// io.EOF means end-of-file was reached gracefully after all
+		// shape-internal reads succeeded, so it's not a reason stop
+		// iterating over all shapes.
+		er.e = nil
+	case er.e != nil:
 		sr.err = fmt.Errorf("Error while reading next shape: %v", er.e)
 		return false
 	}

